controller/src/beerocks/master/db/db.cpp

```diff
@@ -2542,16 +2548,21 @@ bool db::set_client_time_life_delay(const sMacAddr &mac,
 
     auto timestamp = std::chrono::steady_clock::now();
     if (save_to_persistent_db) {
```

Comment:
I wish to indicate to the user that the request to save to the persistent DB was not met since the persistent DB is disabled. how about:
```
// at the begin of each function - 
bool persistent_save = save_to_persistent_db && config.persistent_db;
/* ... */
LOG_IF((save_to_persistent_db && !config.persistent_db), DEBUG) << "persistent db is disabled";
if (persistent_save) { /* ... */ }
```

Reply:
It is the same log each time you check, so I guess one time is enough.
I don't understand why you added the `LOG_IF` _after_ some code. (in your example: `/*...*/`)
For my understanding it should be immediately followed the assignment to `persistent_save` :
```cpp
bool persistent_save = save_to_persistent_db && config.persistent_db;
LOG(DEBUG) << "persistent_save is: " << std::boolalpha << persistent_save << " because save_to_persistent_db is " << save_to_persistent_db << " and config.persistent_db is " << config.persistent_db;
```
This message, for me, is the most useful one. I am not walking around the bush - I have all the information that I need in the log. Looking at the log and looking in the code I don't need to work hard to first translate the log, it is one to one with the code.
Moreover, note that I don't log only when some condition is met, I always log. This kind of logs are beacons in the log, indicating something in the form "was here and with this state". Note however that central places, that happens a lot are not good candidates for such logs.

Reply:
The reason is that when used properly - the API should execute silently and only provide debug logs for valuable information.
The only case where the flow doesn't fully perform what the caller is asking is the case where `save_to_persistent_db` is `true` but the persistent DB is actually disabled  (`config.persistent_db` is `false`) - all other cases are valid.
But I don't want to fail this flow - instead, provide the debug information and perform the runtime DB update.